test/vstest.console.UnitTests/Processors/CollectArgumentProcessorTests.cs

```diff
@@ -586,6 +585,80 @@ public void InitializeXPlatCodeCoverageShouldAddXPlatInProcProcDataCollectoPrope
+        }
+
+        [TestMethod]
+        public void InitializeShouldCreateConfigurationsForNewDataCollectorInRunSettings()
+        {
+            var runsettingsString = string.Format(DefaultRunSettings, "");
+            var runsettings = new RunSettings();
+            runsettings.LoadSettingsXml(runsettingsString);
+            this.settingsProvider.SetActiveRunSettings(runsettings);
+
+            this.executor.Initialize("MyDataCollector;SomeSetting=SomeValue;AnotherSetting=AnotherValue");
+
+            Assert.IsNotNull(this.settingsProvider.ActiveRunSettings);
+            Assert.AreEqual(string.Join(Environment.NewLine,
+                "<?xml version=\"1.0\" encoding=\"utf-16\"?>",
+                "<RunSettings>",
+                "  <RunConfiguration>",
+                "    <TestAdaptersPaths>c:\\AdapterFolderPath</TestAdaptersPaths>",
+                "  </RunConfiguration>",
+                "  <DataCollectionRunSettings>",
+                "    <DataCollectors>",
+                "      <DataCollector friendlyName=\"MyDataCollector\" enabled=\"True\">",
+                "        <Configuration>",
+                "          <SomeSetting>SomeValue</SomeSetting>",
+                "          <AnotherSetting>AnotherValue</AnotherSetting>",
+                "        </Configuration>",
+                "      </DataCollector>",
+                "    </DataCollectors>",
+                "  </DataCollectionRunSettings>",
+                "</RunSettings>"), this.settingsProvider.ActiveRunSettings.SettingsXml);
+        }
+
+        [TestMethod]
+        public void InitializeShouldCreateConfigurationsForExistingDataCollectorInRunSettings()
+        {
+            var runsettingsString = string.Format(DefaultRunSettings,
+                "<DataCollector friendlyName=\"MyDataCollector\" enabled=\"False\">" +
+                "  <Configuration>" +
+                "    <SomeSetting>SomeValue</SomeSetting>" +
+                "  </Configuration>" +
+                "</DataCollector>");
+            var runsettings = new RunSettings();
+            runsettings.LoadSettingsXml(runsettingsString);
+            this.settingsProvider.SetActiveRunSettings(runsettings);
+
+            this.executor.Initialize("MyDataCollector;AnotherSetting=AnotherValue");
+
+            Assert.AreEqual(string.Join(Environment.NewLine,
+                "<?xml version=\"1.0\" encoding=\"utf-16\"?>",
+                "<RunSettings>",
+                "  <RunConfiguration>",
+                "    <TestAdaptersPaths>c:\\AdapterFolderPath</TestAdaptersPaths>",
+                "  </RunConfiguration>",
+                "  <DataCollectionRunSettings>",
+                "    <DataCollectors>",
+                "      <DataCollector friendlyName=\"MyDataCollector\" enabled=\"True\">",
+                "        <Configuration>",
+                "          <SomeSetting>SomeValue</SomeSetting>",
```

Comment:
please add test also when we overwrite value from runsettings

Reply:
Added.